From 97293865b5c3e10fa65009e5f787312a17f47b06 Mon Sep 17 00:00:00 2001
From: Dmitry Shachnev <mitya57@ubuntu.com>
Date: Sun, 27 Sep 2015 15:10:15 +0200
Subject: bitmask: Don't hardcode 64bit size

This looks like an oversight from "quickly testing a potential fix" and
then forgetting to make a production-ready when it works.

https://bugzilla.gnome.org/show_bug.cgi?id=755691

diff --git a/gtk/gtkallocatedbitmask.c b/gtk/gtkallocatedbitmask.c
index f95e312..f655e9d 100644
--- a/gtk/gtkallocatedbitmask.c
+++ b/gtk/gtkallocatedbitmask.c
@@ -312,7 +312,7 @@ _gtk_allocated_bitmask_invert_range (GtkBitmask *mask,
   for (i = start_word; i <= end_word; i++)
     mask->data[i] ^= ALL_BITS;
   mask->data[start_word] ^= (((VALUE_TYPE) 1) << start_bit) - 1;
-  if (end_bit != 63)
+  if (end_bit != VALUE_SIZE_BITS - 1)
     mask->data[end_word] ^= ALL_BITS << (end_bit + 1);
 
   return gtk_allocated_bitmask_shrink (mask);
-- 
cgit v0.10.2

